src/jdk.compiler/share/classes/com/sun/tools/javac/code/Types.java

```diff
@@ -1136,10 +1163,17 @@ public Boolean visitType(Type t, Type s) {
                      return t.hasTag(s.getTag());
                  case TYPEVAR:
                      return isSubtypeNoCapture(t.getUpperBound(), s);
-                 case BOT:
+                 case BOT: {
+                     /* this method can be invoked even from the backend, and warnings can be printed again, so
+                      * make sure that the caller really wants to warn
+                      */
+                     if (s.isNonNullable() && warnStack.head.pos() != null) {
```

Comment:
This is a bit odd. We are defining a type relationship - if "BOT" is not a subtype of T! that is fine, we should just return false. Giving an error seems a bit strange here (I don't think it is normal for subtyping to generate errors).

Reply:
yep you are right